pgjdbc/src/main/java/org/postgresql/jdbc/PgBlobBytea.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2004, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+/*
+ * Implementation of Blob interface backed by Postgres bytea fields.
+ *
+ * See PgBlob.java for implementation using Postgres Large Objects.
+ */
+
+package org.postgresql.jdbc;
+
+import org.postgresql.util.GT;
+import org.postgresql.util.PSQLException;
+import org.postgresql.util.PSQLState;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.sql.Blob;
+import java.sql.SQLException;
+import java.util.Arrays;
+
+public class PgBlobBytea implements java.sql.Blob {
+
+  private byte[] data;
+
+  public static final int MAX_BYTES = 1073741824; // 1GiB max size of a bytea
+
+  public PgBlobBytea() {
+    this.data = new byte[0];
+  }
+
+  public PgBlobBytea(byte[] in) throws SQLException {
+    if (in.length > MAX_BYTES) {
+      throw new PSQLException(GT.tr("input data too long for bytea type Blob {0}", in.length),
+                              PSQLState.INVALID_PARAMETER_VALUE);
+    }
+    this.data = in;
+  }
+
+  // This method frees the Blob object and releases the resources that it holds.
+  @Override
+  public void	free() 	 {
+    this.data = new byte[0];
+  }
+
+  // Retrieves the BLOB value designated by this Blob instance as a stream.
+  @Override
+  public InputStream getBinaryStream() {
+    if (this.data != null) {
+      return new ByteArrayInputStream(this.data);
+    }
+    return new ByteArrayInputStream(new byte[0]);
+  }
+
+  // Returns an InputStream object that contains a partial Blob value,
+  // starting with the byte specified by pos, which is length bytes in length.
+  @Override
+  public InputStream getBinaryStream​(long pos, long length) throws SQLException {
+    if (this.data == null) {
```

Comment:
How would data ever be null ?

Reply:
see previous comment  re commit c8c1bbf1